### ISSUE TITLE
Typescript error in multi-repo setup

### ISSUE THREAD

**Issue by silveltman** [withastro/astro#8616]
### Astro Info

```block
Astro                    v3.1.1
Node                     v18.17.1
System                   macOS (arm64)
Package Manager          yarn
Output                   server
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/svelte
                         @astrojs/sitemap
                         @astrojs/prefetch
                         astro-robots-txt
                         unocss:config
                         unocss:transformers:default
                         unocss:transformers:pre
                         unocss:transformers:post
                         unocss:devtools
                         unocss:api
                         unocss:inspector
                         unocss:global:build:scan
                         unocss:global:content
                         unocss:global:build:generate
                         unocss:global:build:bundle
                         unocss:global
                         unocss:global:post
```

### If this issue only occurs in one browser, which browser is a problem?

vscode

### Describe the Bug

The following typescript error when importing svelte components into astro files:

```
File '.../astro/src/components/MyComponent.svelte.tsx' is not a module.

Is the `@astrojs/svelte` package installed? You can add it to your project by running the following command: `astro add svelte`. If already installed, restarting the language server might be necessary in order for the change to take effect.ts(230
```

I'm using a multi-repo setup where I have the root, and within this root are 2 folder, of which one is 'astro'. When opening the astro folder itself, the errors disappear.

When using yarn workspaces the problem disappeard, but I need the node_modules to be installed in the astro folder itself

### What's the expected result?

Typescript to recognize the fact that @astrojs/svelte is installed.

### Link to Minimal Reproducible Example

Can't because its a vscode error

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by TClark1011** [comment]
I'm getting the same thing in my [project](https://github.com/TClark1011/my-dnd-homebrew-site) and it is not a multi repo project. Its occurring in the `/src/pages/homebrew/[slug].astro` file when trying to import `$/components/SlideOutPanel.svelte`, I am using the `paths` field in `tsconfig` for that aliasing, and the error still occurred for me even if I switched to a regular relative path import. I am using bun as the package manager, but I have tried to switching to yarn and the error did not go away.
